app/view_models/Amount.scala

```diff
@@ -39,6 +39,8 @@ case class Amount(amount: BigDecimal, currency: String) {
 
   def isZero = amount == BigDecimal(0)
 
+  def isZeroOrLess = amount <= BigDecimal(0)
```

Comment:
Then this becomes: 

```
  def isZeroOrLess = isZero  || amount < BigDecimal(0)
```

it's odd but it's how floating point comparison works. You can't guarantee equality, you can test this by:

```
val a = 0.3
val b = 0.1 + 0.2.  // will be 0.300000000004 or something close

a == b (false)
```